### PR TITLE
variables update their value when set again

### LINKED ISSUE #4255: variable overshadowing or proper handling of variables when they are set multiple times
### Description

let variables that are set multiple times in config use the value that was last set

### PR DESCRIPTION
**Pull request by flick0**
#### Describe your PR, what does it fix/add?
fixes #4255

```
$var=1
// value of var is now '1'
$var=2
// value of var is now '2'
$var=3
// value of var is now '3'
```

### CODE REVIEW

File: src/config/ConfigManager.cpp [hunk 1]
```diff
@@ -390,9 +390,21 @@ void CConfigManager::configSetValueSafe(const std::string& COMMAND, const std::s
...
+                        found = true;
+                    }
+                }
 
+                if (!found) {
+                    Debug::log(LOG, "Registered dynamic var \"{}\" -> {}", COMMAND, VALUE);
+                    configDynamicVars.emplace_back(std::make_pair<>(COMMAND.substr(1), VALUE));
+                }
+                
                 std::sort(configDynamicVars.begin(), configDynamicVars.end(), [&](const auto& a, const auto& b) { return a.first.length() > b.first.length(); });
```
Comment (vaxerski) [inline src/config/ConfigManager.cpp:408]: sort can be moved to !found, if we don't add a new var sorting is redundant.

**Comment by vaxerski** [comment]
oops code style fails. run clang-format

**Comment by flick0** [comment]
done

**Review by vaxerski — APPROVED** [review]
fwiw hyprlang already does this

**Pull request merged**